### PR TITLE
Tanner

### PR DESCRIPTION
**Pull request by TannerDrayton**

### CODE REVIEW

File: LoggingKata.Test/TacoParserTests.cs [hunk 1]
```diff
@@ -1,29 +1,58 @@
...
+using System;
+using Xunit;
+using LoggingKata;
+
+namespace LoggingKata.Test
+{
+    public class TacoParserTests
+    {
+        [Fact]
+        public void ShouldDoSomething()
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:10]: This method is empty and could/should be removed

File: LoggingKata.Test/TacoParserTests.cs [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
+            TacoParser TacoParse = new TacoParser();
+            Point TacoBellLocation = new Point(TestLatitude, TestLongitude);
+
+            // Act
+            ITrackable actual = TacoParse.Parse(str);
+
+            // Assert
+            Assert.Equal(TestName, actual.Name);
+            Assert.Equal(TacoBellLocation.Latitude, actual.Location.Latitude);
+            Assert.Equal(TacoBellLocation.Longitude, actual.Location.Longitude);
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:37]: Nice assertions and use of inline data 👍

File: LoggingKata.Test/TacoParserTests.cs [hunk 3]
```diff
@@ -1,29 +1,58 @@
...
+            // Act
+
+
+            // Assert
+        }
+
+        [Theory]
+        [InlineData("34.996237,-85.291147,Taco Bell Chattanooga", "Taco Bell Chattanooga", 34.996237, -85.291147)]
+        [InlineData("32.555148,-84.946447,Taco Bell Columbus", "Taco Bell Columbus", 32.555148, -84.946447)]
+
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:24]: This empty line could be removed

File: LoggingKata.Test/TacoParserTests.cs [hunk 4]
```diff
@@ -1,29 +1,58 @@
...
+            Assert.Equal(TacoBellLocation.Longitude, actual.Location.Longitude);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("It's, Taco, Tuesdaaaaaay")]
+        [InlineData("34.9962, NoLongitude ,Taco Bell Chattanooga")]
+        public void ShouldFailParse(string str)
+        {
+            // TODO: Complete Should Fail Parse
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:46]: This is done now, so the "TODO" should be removed

File: LoggingKata/TacoLogger.cs [hunk 5]
```diff
@@ -28,5 +28,10 @@ public void LogDebug(string log)
         {
             Console.WriteLine($"Debug: {log}");
         }
+
+        public void LogError()
+        {
+            throw new NotImplementedException();
```
Comment (chipelmer) [inline LoggingKata/TacoLogger.cs:34]: It's interesting that you modified the ILog interface to add this method, then added the method but didn't implement it

File: LoggingKata/TacoParser.cs [hunk 6]
```diff
@@ -1,18 +1,47 @@
...
         readonly ILog logger = new TacoLogger();
-        
+
         public ITrackable Parse(string line)
         {
             logger.LogInfo("Begin parsing");
+            TacoBell tacoBell = new TacoBell();
+
+            double longitude;
+            double latitude;
```
Comment (chipelmer) [inline LoggingKata/TacoParser.cs:14]: These are declared kind of far from where they are used.  Consider this on lines 30 and 33:
```CS
bool ConvertLongitude = double.TryParse(LongitudeAsString, out double longitude);
bool ConvertLatitude = double.TryParse(LatitudeAsString, out double latitude);
```

File: LoggingKata.Test/TacoParserTests.cs [hunk 7]
```diff
@@ -1,29 +1,58 @@
...
+            // Assert
+            Assert.Equal(TestName, actual.Name);
+            Assert.Equal(TacoBellLocation.Latitude, actual.Location.Latitude);
+            Assert.Equal(TacoBellLocation.Longitude, actual.Location.Longitude);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("It's, Taco, Tuesdaaaaaay")]
+        [InlineData("34.9962, NoLongitude ,Taco Bell Chattanooga")]
```
Comment (chipelmer) [inline LoggingKata.Test/TacoParserTests.cs:43]: For extra validation, consider some more edge cases:
```CS
"10, ten, Taco Bell"
"ten, 10, Taco Bell"
"10, 10"
"10, 10, "
"10,,Taco Bell"
",10, Taco Bell"
", , Taco Bell"
"10, Taco Bell, 10"
"Taco Bell, 10, 10"
"91, 10, Taco Bell" // above max latitude
"-91, 10, Taco Bell" // below min latitude
"10, 181, Taco Bell" // above max longitude
"10, -181, Taco Bell" // below min longitude
```
These will force your parse method to make some additional checks, though

**Review by chipelmer — APPROVED** [review]
This looks very good, well done 👍
